### ISSUE TITLE
Terraform 0.12 compatibility

### ISSUE THREAD

**Issue by grogancolin** [terraform-providers/terraform-provider-cloudflare#259]
I was trying out the new terraform 0.12 beta, and notice the Cloudflare provider is as yet incompatible.

Is there a plan to upgrade it?

```
$ terraform init

No available provider "cloudflare" plugins are compatible with this Terraform version.

From time to time, new Terraform major releases can change the requirements for
plugins such that older plugins become incompatible.

Terraform checked all of the plugin versions matching the given constraint:
    ~> 1.11,~> 1.12

Unfortunately, none of the suitable versions are compatible with this version
of Terraform. If you have recently upgraded Terraform, it may be necessary to
move to a newer major release of this provider. Alternatively, if you are
attempting to upgrade the provider to a new major version you may need to
also upgrade Terraform to support the new version.

Terraform v0.12.0-beta1 was released before all providers were updated for
compatibility. For more information on provider availability for beta1,
including workarounds for testing with not-yet-supported providers, please see
the release announcement:
    https://www.hashicorp.com/blog/announcing-terraform-0-1-2-beta1
```

**Comment by jacobbednarz** [comment]
Supporting Terraform 0.12 is planned however the release hasn't fully stabilised yet (see the Terraform [milestone](https://github.com/hashicorp/terraform/milestone/5)). Once it's closer, there will be some focus on it.

**Comment by grogancolin** [comment]
Thanks, good to know.

**Comment by garrettgalow** [comment]
beta1 just got released today cc @patryk 

https://www.hashicorp.com/blog/announcing-terraform-0-1-2-beta1

**Comment by iandelahorne** [comment]
Hi! We use the cloudflare provider a fair bit and would love to be able to start testing the upgrade path now that beta 2 is out:

https://www.hashicorp.com/blog/announcing-terraform-0-1-2-beta-2

**Comment by jacobbednarz** [comment]
Work to support 0.12 has been started in #309 already.

**Comment by jacobbednarz** [comment]
0.12 compatibility was merged this morning. Release is expected in the next couple of days.

**Comment by rafilkmp3** [comment]
can I build myself the provider to avoid this ?
```
No available provider "cloudflare" plugins are compatible with this Terraform version.

From time to time, new Terraform major releases can change the requirements for
plugins such that older plugins become incompatible.

Terraform checked all of the plugin versions matching the given constraint:
    (any version)

Unfortunately, none of the suitable versions are compatible with this version
of Terraform. If you have recently upgraded Terraform, it may be necessary to
move to a newer major release of this provider. Alternatively, if you are
attempting to upgrade the provider to a new major version you may need to
also upgrade Terraform to support the new version.

Consult the documentation for this provider for more information on
compatibility between provider versions and Terraform versions.

```

**Comment by jacobbednarz** [comment]
Yes, please see [building the provider](https://github.com/terraform-providers/terraform-provider-cloudflare#building-the-provider) section of the project README.

**Comment by jacobbednarz** [comment]
1.15.0 was released today with 0.12 compatibility.

**Comment by rafilkmp3** [comment]
> Yes, please see [building the provider](https://github.com/terraform-providers/terraform-provider-cloudflare#building-the-provider) section of the project README.

Thanks, im able to build but much more happier to use released version, thanks for your time
